### ISSUE TITLE
Make it possible to view and debug the lifecycle of a secret populated by the kubernetes_secrets provider

### ISSUE THREAD

**Issue by cmacknz** [elastic/elastic-agent#6187]
The agent kubernetes_secrets provider has an internal cache that updates periodically and includes a TTL. https://www.elastic.co/guide/en/fleet/current/kubernetes_secrets-provider.html

The secrets provider includes limited logging about when individual secrets are updated or expired. The values populated by this provider should also be considered secrets in diagnostics and be redacted, making it challenging or impossible to tell if a secret updated when it was supposed to. 

When the cache updates today for example, only a single log line indicating that some unindicated secret value was updated or deleted without specifying which is logged https://github.com/elastic/elastic-agent/blob/56901edc3f2c6d1fc0c1271fad374cdbb24ed7f2/internal/pkg/composable/providers/kubernetessecrets/kubernetes_secrets.go#L112-L116

We recently had an internal case where a JWT token populated as a kubernetes secret in the agent policy did not rotate as expected, and the limited logging in agent made it impossible to tell if agent was involved in the root cause.

Add logging that would allow us to verify that agent is updating secrets at the expected times with the correct values in the policy, without leaking the actual secret values.

**Comment by adamkasztenny** [comment]
Maybe the hash of the secret could be logged? Elasticsearch does something similar for the JWKS [here](https://github.com/elastic/elasticsearch/pull/113454/files#diff-ebfd160fbdd616be6a5a2d8ec86bee49664235f52e69945b3bf8715c5a1cdbd7R150) (although admittedly that's not necessarily secret data, the JWKS could be publicly accessible). This could be an optional setting that is disabled by default.

**Comment by pkoutsovasilis** [comment]
@adamkasztenny 👋 instead of hashing actual secret data, which I am entirely comfortable with doing 😅, would logging the resource version result in the same benefits as hashing?

**Comment by adamkasztenny** [comment]
@pkoutsovasilis Yep that's a good idea, I think that's better than using the hash and should achieve the same thing. We just care if the secret has changed, not the contents.
